src/NLog/Fluent/Log.cs

```diff
@@ -117,15 +117,26 @@ public static LogBuilder Fatal([CallerFilePath]string callerFilePath = null)
 
         private static LogBuilder Create(LogLevel logLevel, string callerFilePath)
         {
-            string name = Path.GetFileNameWithoutExtension(callerFilePath ?? string.Empty);
-            var logger = string.IsNullOrWhiteSpace(name) ? _logger : LogManager.GetLogger(name);
-
+            var logger = GetLogger(callerFilePath);
             var builder = new LogBuilder(logger, logLevel);
             if (callerFilePath != null)
                 builder.Property("CallerFilePath", callerFilePath);
 
             return builder;
         }
+
+        private static ILogger GetLogger(string callerFilePath)
+        {
+            try
+            {
+                string name = Path.GetFileNameWithoutExtension(callerFilePath ?? string.Empty);
+                return string.IsNullOrWhiteSpace(name) ? _logger : LogManager.GetLogger(name);
+            }
+            catch
+            {
+                return _logger;
```

Comment:
Maybe an internal log (trace or debug)?

Reply:
Fixed to match `InternalLogger.Warn(ex, "Error when formatting a message.");`